### PR TITLE
waller and soccer mom, github wasn't working before so soccer mom is …

### PR DESCRIPTION
**Pull request by shourikb**
Completes RC SW tutorials

### CODE REVIEW

**Comment by shourikb** [comment]
Screenshot 2023-09-17 at 7 32 39 PM

**Comment by shourikb** [comment]
Screenshot 2023-10-01 at 7 36 18 PM
Screenshot 2023-10-01 at 7 36 31 PM

**Comment by shourikb** [comment]
@sid-parikh 

Soccer mom could be quite confusing since I needed to combine things from multiple classes but it wasn't clear what was needed and what wasn't. For the runner part, it could also be a little confusing since the documentation for the methods we were calling wasn't provided so the number of parameters required to be passed was a little confusing

File: soccer/src/soccer/soccermom.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+void SoccerMom :: send() {
+  {
+    auto message = std_msgs::msg::String();
+    if (is_blue_) {
+        message.data = "Blueberry";
+    } else {
+        message.data = "Banana";
+    }
+    RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
```
Comment (sid-parikh) [inline soccer/src/soccer/soccermom.cpp:25]: Prefer SPDLOG_INFO

File: soccer/src/soccer/soccermom.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#include "soccermom.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace tutorial {
+
+SoccerMom :: SoccerMom() : Node("soccer_mom") {
+       publisher_ = this->create_publisher<std_msgs::msg::String>("team_fruit", 10);
+       subscription_ = this->create_subscription<rj_msgs::msg::TeamColor>(
+      "referee/team_color", 10, [this](rj_msgs::msg::TeamColor::SharedPtr color) {
```
Comment (sid-parikh) [inline soccer/src/soccer/soccermom.cpp:10]: set ros QoS setting to 1 unless you have a specific reason to use 10

File: soccer/src/soccer/soccermom.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          is_blue_ = color->is_blue;
+          });
+       timer_ = create_wall_timer(std::chrono::milliseconds(16), [this]() { send(); });
```
Comment (sid-parikh) [inline soccer/src/soccer/soccermom.cpp:13]: interesting implementation. a timer is fine, but in this case it is best to only publish in reaction to the subscription, as opposed to every 16 ms. you only need to publish when the data has changed.

File: soccer/src/soccer/strategy/agent/agent_action_client.cpp [hunk 4]
```diff
@@ -170,13 +174,13 @@ void AgentActionClient::update_position(const rj_msgs::msg::PositionAssignment::
             next_position_ = std::make_unique<Goalie>(robot_id_);
             break;
         case 1:
-            next_position_ = std::make_unique<Defense>(robot_id_);
```
Comment (sid-parikh) [inline soccer/src/soccer/strategy/agent/agent_action_client.cpp:173]: this isn't correct. this switch-case must correspond exactly to the Position enum in Coach Node.

File: soccer/src/soccer/strategy/agent/position/runner.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    State next_state = current_state_;
+    // handle transitions between current state
+    WorldState* world_state = this->world_state();
+
+    rj_geometry::Point robot_position = world_state->get_robot(true, robot_id_).pose.position();
+  
+     if (current_state_ == SIDE1) {
+      rj_geometry::Point end_position = rj_geometry::Point{3.0, 3.0};
+      double distance_to_end = robot_position.dist_to(end_position);
+      if (distance_to_end < 0.1) {
```
Comment (sid-parikh) [inline soccer/src/soccer/strategy/agent/position/runner.cpp:25]: thresholds are arbitrary. prefer using check_is_done() to see if a previous motion command has completed. if you actually need different behavior, leave a comment explaining why.

File: soccer/src/soccer/strategy/agent/position/runner.hpp [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    //    communication::AgentPosRequestWrapper request) override;
+
+    // void derived_acknowledge_pass() override;
+    // void derived_pass_ball() override;
+    // void derived_acknowledge_ball_in_transit() override;
+    
+private:
+    std::optional<RobotIntent> derived_get_task(RobotIntent intent) override;
+    
+    enum State {
```
Comment (sid-parikh) [inline soccer/src/soccer/strategy/agent/position/runner.hpp:35]: prefer `enum class` to a plain `enum`

File: soccer/src/soccer/strategy/coach/coach_node.hpp [hunk 7]
```diff
@@ -35,12 +36,12 @@ enum MatchSituation {
...
 };
 
-enum Positions { Goalie, Defense, Offense, PenaltyPlayer, GoalKicker };
+enum Positions { Goalie, Defense, Offense, Runner, PenaltyPlayer, GoalKicker };
 
 // These values are explicitly declared because they are the ints that are published to
 // strategy/positions i.e. the same values as strategy::Positions
 namespace OverridePosition {
-enum OverridePosition { Goalie = 0, Defense = 1, Offense = 2, None = 3 };
+enum OverridePosition { Goalie = 0, Defense = 3, Offense = 2, Runner = 1, None = 4 };
```
Comment (sid-parikh) [inline soccer/src/soccer/strategy/coach/coach_node.hpp:44]: changing this doesn't do anything; it's for the UI tab to assign positions (which is broken anyway, LOL)

**Review by sid-parikh — APPROVED** [review]
Good work! Take a look at my feedback and then *close* this PR.

**Pull request closed without merging**